bindings/node/test/clientEncryption.test.js

```diff
@@ -186,3 +189,143 @@ describe('ClientEncryption', function() {
+      const client = this.client;
+      this.client = undefined;
+      return client.close();
+    }
+  });
+
+  function makeOptions(keyAltNames) {
+    return Object.assign({}, dataKeyOptions, { keyAltNames });
+  }
+
+  describe('errors', function() {
+    [42, 'hello', { keyAltNames: 'foobar' }, /foobar/].forEach(val => {
+      it(`should fail if typeof keyAltNames = ${typeof val}`, function(done) {
+        try {
+          this.encryption.createDataKey('aws', makeOptions(val), () => {
+            done(new Error('expected test to fail'));
+          });
+        } catch (e) {
+          try {
+            expect(e).to.be.an.instanceof(TypeError);
+            done();
+          } catch (_e) {
+            done(_e);
+          }
+        }
+      });
+    });
+
+    [undefined, null, 42, { keyAltNames: 'foobar' }, ['foobar'], /foobar/].forEach(val => {
+      it(`should fail if typeof keyAltNames[x] = ${typeof val}`, function(done) {
+        try {
+          this.encryption.createDataKey('aws', makeOptions([val]), () => {
+            done(new Error('expected test to fail'));
+          });
+        } catch (e) {
+          try {
+            expect(e).to.be.an.instanceof(TypeError);
+            done();
+          } catch (_e) {
+            done(_e);
+          }
+        }
+      });
+    });
+  });
+
+  it('should create a key with keyAltNames', function() {
+    let dataKey;
+    return this.encryption
+      .createDataKey('aws', makeOptions(['foobar']))
+      .then(_dataKey => (dataKey = _dataKey))
+      .then(() => this.collection.findOne({ keyAltNames: 'foobar' }))
+      .then(document => {
+        expect(document)
+          .to.have.property('keyAltNames')
+          .that.includes.members(['foobar']);
+        expect(document).to.have.nested.property('_id.buffer');
+        expect(Buffer.compare(document._id.buffer, dataKey.buffer)).to.equal(0);
+      });
+  });
+
+  it('should create a key with multiple keyAltNames', function() {
+    let dataKey;
+    let doc1;
+    let doc2;
+    return this.encryption
+      .createDataKey('aws', makeOptions(['foobar', 'fizzbuzz']))
+      .then(_dataKey => (dataKey = _dataKey))
+      .then(() => this.collection.findOne({ keyAltNames: 'foobar' }))
+      .then(document => (doc1 = document))
+      .then(() => this.collection.findOne({ keyAltNames: 'fizzbuzz' }))
+      .then(document => (doc2 = document))
+      .then(() => {
+        expect(doc1)
+          .to.have.property('keyAltNames')
+          .that.includes.members(['foobar', 'fizzbuzz']);
+        expect(doc1).to.have.nested.property('_id.buffer');
+        expect(Buffer.compare(doc1._id.buffer, dataKey.buffer)).to.equal(0);
+        expect(doc2)
+          .to.have.property('keyAltNames')
+          .that.includes.members(['foobar', 'fizzbuzz']);
+        expect(doc2).to.have.nested.property('_id.buffer');
+        expect(Buffer.compare(doc2._id.buffer, dataKey.buffer)).to.equal(0);
```

Comment:
nit: chai supports buffer comparison with `eql`: `expect(doc2._id.buffer).to.eql(dataKey.buffer)`

Reply:
really? does that also work with a `deepEquals`? I've been writing myself in knots for this :(

Reply:
yeah!